server/src/main/java/org/opensearch/common/settings/IndexScopedSettings.java

```diff
@@ -182,7 +182,7 @@ public final class IndexScopedSettings extends AbstractScopedSettings {
                 BitsetFilterCache.INDEX_LOAD_RANDOM_ACCESS_FILTERS_EAGERLY_SETTING,
                 IndexModule.INDEX_STORE_TYPE_SETTING,
                 IndexModule.INDEX_STORE_PRE_LOAD_SETTING,
-                IndexModule.INDEX_STORE_HYBRID_MMAP_EXTENSIONS,
+                IndexModule.INDEX_STORE_NIO_EXTENSIONS,
```

Comment:
It looks to me this change will be a nightmare to migrate for anyone who uses `INDEX_STORE_HYBRID_MMAP_EXTENSIONS` at the moment, plus we just cannot delete this setting otherwise migration from 2.x to 3.x won't be possible I believe (the setting won't be recognized).

Reply:
Yes I agree this will disrupt users. However this change is meant to safeguard against future regressions.

Is there any mechanism for deleting these type of index settings? If not, it will be difficult to solve this. We could keep both settings and only use `INDEX_STORE_HYBRID_MMAP_EXTENSIONS` if the user has set it. Although may be overcomplicating.

Reply:
> Yes I agree this will disrupt users. However this change is meant to safeguard against future regressions.

I think Lucene very rarely adds new formats, but in any case, we could try to catch that internally (to safeguard against future regressions). For example, we enumerate all MMAP / NIO extensions (as you did) and assert when new extension is encountered that is not in both sets.

Reply:
> For example, we enumerate all MMAP / NIO extensions (as you did) and assert when new extension is encountered that is not in both sets.

@reta - Thats a good suggestion, although not sure how can validate whenever new extension is encountered. Ideally, you want to do within unit/integration test which I am not sure how can we write.

> It looks to me this change will be a nightmare to migrate for anyone who uses INDEX_STORE_HYBRID_MMAP_EXTENSIONS at the moment, 

Maybe we can do the change such that for anyone using this setting already, the NIO list is automatically populated as the differential of MMAP setting and the MMAP setting is eventually deprecated?

> plus we just cannot delete this setting otherwise migration from 2.x to 3.x won't be possible I believe (the setting won't be recognized)

If I remember correctly, unrecognized settings get archived?

Reply:
Thanks @jainankitk 

> Ideally, you want to do within unit/integration test which I am not sure how can we write.

Correct, or we could use `opensearch-benchmarks` for that: it includes most (if not all) data types and very likely the problem would manifest here.

> Maybe we can do the change such that for anyone using this setting already, the NIO list is automatically populated as the differential of MMAP setting and the MMAP setting is eventually deprecated?

That could be an option, yes

> If I remember correctly, unrecognized settings get archived?

Yes. you are correct (and I was not). the only risk than would be ignoring user settings if any were provided .... But in any case - we should start with deprecation first, not removal.

Reply:
Based on our discussion I will make the following changes:
1. Label `INDEX_STORE_HYBRID_MMAP_EXTENSIONS` for deprecation
2. Keep both `INDEX_STORE_HYBRID_NIO_EXTENSIONS`, `INDEX_STORE_HYBRID_MMAP_EXTENSIONS`
3. If user sets `INDEX_STORE_HYBRID_NIO_EXTENSIONS` it will take priority over `INDEX_STORE_HYBRID_MMAP_EXTENSIONS`
4. Users which use `INDEX_STORE_HYBRID_MMAP_EXTENSIONS` will see no change (unless overridden by 3)

New filetypes will use mmap even when the user does not specify it.

[Future TODO] Remove `INDEX_STORE_HYBRID_MMAP_EXTENSIONS` completely in a later release

Reply:
Sounds good to me, thanks @dzane17